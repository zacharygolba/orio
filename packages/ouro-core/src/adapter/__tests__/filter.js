// @flow

import Filter from '../filter'
import * as ouro from '../../'

let subj

beforeEach(() => {
  const { producer } = ouro.of(1, 2, 3)
  subj = new Filter(producer, value => value > 1)
})

test('#@@iterator()', () => {
  for (const item of subj) {
    expect(item).toMatchSnapshot()
  }
})

test('#next()', () => {
  expect(subj.next()).toMatchSnapshot()
  expect(subj.next()).toMatchSnapshot()
  expect(subj.next()).toMatchSnapshot()
  expect(subj.next()).toMatchSnapshot()
})
