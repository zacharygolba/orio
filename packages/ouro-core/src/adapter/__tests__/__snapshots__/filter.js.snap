// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#@@iterator() 1`] = `2`;

exports[`#@@iterator() 2`] = `3`;

exports[`#next() 1`] = `
Object {
  "done": false,
  "value": 2,
}
`;

exports[`#next() 2`] = `
Object {
  "done": false,
  "value": 3,
}
`;

exports[`#next() 3`] = `
Object {
  "done": true,
  "value": undefined,
}
`;

exports[`#next() 4`] = `
Object {
  "done": true,
  "value": undefined,
}
`;
